text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,365 @@
+* **Flattened Transaction Structure:** To reduce processing overhead associated with serialization/deserialization of deeply nested structures, Fabric-X adopts a flatter transaction model (see "Reference-level explanation" for details).
+* **Sequential Versioning:** Fabric-X adopts sequential versioning of key/values. This allows endorsers to predict the next version without querying the database, potentially speeding up endorsement.
+
+**Impact on Fabric Users:**
+
+* **Deployment:** Deployment scripts and tools will need to be adapted for a microservice environment (e.g., using Kubernetes or similar orchestration platforms). Each organization will deploy its set of microservices and its own distributed database instance. The minimum number of logical components per organization is ten (1 endorser, router, batcher, consenter, assembler, sidecar, coordinator, signature/policy verifier, validator-committer, and a single database node cluster), though these can be co-hosted for smaller setups. For BFT consensus in Arma, a minimum of 4 Consenter nodes (to tolerate F=1 faulty node) distributed across participating organizations is required.
+* **Configuration:** Configuration will be more granular, specific to each microservice. The genesis configuration will use the same config block format and `configtxgen` (or equivalent), with additional fields for the new ordering services. MSP, ACLs, and Policies remain conceptually the same.
+* **Debugging:** Troubleshooting will involve analyzing logs and interactions across multiple services.
+* **Performance:** Expect significantly higher throughput (demonstrated >150,000 TPS) and lower latency.
+* **Feature Set:** Be aware that Fabric-X will not initially support all HLF v2.x/v3.x features (e.g., Private Data Collections, Channels). The focus is on core transactional performance. A roadmap for reintroducing essential Fabric features will be developed post-initial release.
+* **Security:** While individual components are more isolated, the overall security of the distributed system, including inter-service communication, must be carefully managed. The compromise of a Query Service node does not inherently allow it to substitute data in the sharded distributed database if the database itself maintains strong consistency and write integrity is handled by the VCs; the Query Service primarily performs reads. Writes to the database are governed by the VCs after successful validation.
+* **Programming Model Shift:** Developers will transition from chaincode to 'Flows' using FSC or an endorser SDK. This requires learning but offers more power and flexibility.
+
+**Migration and Coexistence Strategy**
```

Comment:
I think this section should be compiled into some initial roadmap, clearly outlining the roadmap demonstrating intent to support the community with migration, and some means of interoperability; otherwise, it's just a list enumerating some future intent, highlighting some possibility of the migration.

Reply:
No roadmap would be provided. We need to gather feedback on what features are needed and what are not. It is going to be a long discussion over years.